lib/src/base_client.dart

```diff
@@ -13,136 +13,46 @@ import 'exception.dart';
 import 'request.dart';
 import 'response.dart';
 
-/// The abstract base class for an HTTP client. This is a mixin-style class;
-/// subclasses only need to implement [send] and maybe [close], and then they
-/// get various convenience methods for free.
+/// The abstract base class for an HTTP client.
+///
+/// This is a mixin-style class; subclasses only need to implement [send] and
```

Comment:
"Mixin-style" is confusing now that Dart has actual mixins. I'd just remove that bit.